tic/registers.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package tic
+
+import (
+	"encoding/binary"
+)
+
+// getVar8 reads an 8 bit value from the Tic at a given register offset.
+func (d *Dev) getVar8(offset Offset) (uint8, error) {
+	const length = 1
+	buffer, err := d.getSegment(cmdGetVariable, offset, length)
+	if err != nil {
+		return 0, err
+	}
+
+	return buffer[0], nil
+}
+
+// getVar16 reads a 16 bit value from the Tic at a given register offset.
+func (d *Dev) getVar16(offset Offset) (uint16, error) {
+	const length = 2
+	buffer, err := d.getSegment(cmdGetVariable, offset, length)
+	if err != nil {
+		return 0, err
+	}
+
+	return binary.LittleEndian.Uint16(buffer), nil
+}
+
+// getVar32 reads a 32 bit value from the Tic at a given register offset.
+func (d *Dev) getVar32(offset Offset) (uint32, error) {
+	const length = 4
+	buffer, err := d.getSegment(cmdGetVariable, offset, length)
+	if err != nil {
+		return 0, err
+	}
+
+	return binary.LittleEndian.Uint32(buffer), nil
+}
+
+// commandQuick sends a command without additional data.
+func (d *Dev) commandQuick(cmd command) error {
+	writeBuf := []byte{uint8(cmd)}
+	err := d.c.Tx(writeBuf, nil)
+	return err
+}
+
+// commandW7 sends a command with a 7 bit value. The MSB of val is ignored.
+func (d *Dev) commandW7(cmd command, val uint8) error {
+	writeBuf := []byte{byte(cmd), val & 0x7F}
+	err := d.c.Tx(writeBuf, nil)
+	return err
+}
+
+// commandW32 sends a command with a 32 bit value.
+func (d *Dev) commandW32(cmd command, val uint32) error {
+	writeBuf := make([]byte, 5)
```

Comment:
FYI, this works:
```
writeBuf := [5]byte{byte(cmd)}
```

The only drawback is that line 64 would have to be replaced with:

```
err := d.c.Tx(writeBuf[:], nil)
```

Reply:
[577d57f](https://github.com/periph/devices/pull/78/commits/577d57f1b0243ac545e2eaf81ffd6589f6503697)

Reply:
Personally I wouldn't have changed the others, since the number of elements desired in the array matched the number of elements provided. But it's okay.